R/add_cols.R

```diff
@@ -118,60 +118,97 @@ NULL
 }
 
 #' @name .add_date
-.add_deaths <- function(.data,
-                        onset_to_death,
-                        hosp_death_risk,
-                        non_hosp_death_risk) {
+.add_outcome <- function(.data,
+                         onset_to_death,
+                         onset_to_recovery,
+                         hosp_death_risk,
+                         non_hosp_death_risk,
+                         config) {
   infected_idx <- .data$infected == "infected"
   num_infected <- sum(infected_idx)
-  .data$deaths <- NA_real_
-  .data$deaths[infected_idx] <- .data$time[infected_idx] +
-    onset_to_death(num_infected)
+  .data$outcome <- "contact"
+  .data$outcome_time <- NA_real_
+  .data$outcome[infected_idx] <- "recovered"
+  .data$outcome_time[infected_idx] <- .data$time[infected_idx] +
+    onset_to_recovery(num_infected)
+  hosp_idx <- !is.na(.data$hospitalisation) & infected_idx
+  non_hosp_idx <- is.na(.data$hospitalisation) & infected_idx
 
-  apply_death_risk <- function(.data, risk, hosp = TRUE) {
+  # define functions used only by .add_outcome
+  time_varying_risk <- function(.data, died_idx, config) {
+    # check time-varying function
+    .check_func_req_args(config$time_varying_death_risk)
+
+    # sample which deaths to keep with time-varying risk
+    prob_death <- config$time_varying_death_risk(.data$time[died_idx])
+    max_prob_death <- stats::optimise(
+      f = config$time_varying_death_risk,
+      interval = c(0, 100),
+      maximum = TRUE
+    )$objective
```

Comment:
I understand what `stats::optimise()` is doing here, but a comment explaining it would be good. Something as simple as, "determine maximum possible value of the time-dependent CFR function".

Reply:
Resolved in e39a0dc5e812aed46b896f43f2c42c9d43cc0c81.